queue/init.lua

```diff
@@ -6,6 +6,9 @@ local function register_driver(driver_name, tube_ctr)
         error('tube control methods must contain functions "create_space"'
               .. ' and "new"')
     end
+    if queue.driver[driver_name] then
+        error(('overriding registered driver "%s"'):format(driver_name))
+    end
```

Comment:
So before box.cfg() we'll not receive an error on `register_driver()` on attempt to override a built-in dirver (it'll be postponed until box.cfg() call). But we'll get the error when will try the same after box.cfg().

For me it looks a bit lopsided and I would see whether moving `queue.driver` table from `abstract.lua` to `init.lua` may help. It would also remove 'overriding default driver' check from the box.cfg wrapper, which should be done right in `register_driver()`, IMHO.

Reply:
Updated.